### ISSUE TITLE
Document how metadata works

### ISSUE THREAD

**Issue by JoanZapata** [commanded/commanded#104]
On the projection side, the metadata received can look like this:

```
%{:created_at => ~N[2017-10-30 11:19:56.178901], :event_number => 1,
  :stream_id => "e42a588d-2cda-4314-a471-5d008cce01fc", :stream_version => 1,
  "command_uuid" => "1c4d0e09-2958-41db-bce0-0fad4d644825"}
```

As seen with @slashdotdash, it should be documented that commanded metadata have `atom` keys, while user provided metadata have `String` keys. This is because the user metadata is serialized/deserialized from JSON.

**Comment by astery** [comment]
But should it be better to have consistently only atoms or strings keys despite of origin?

**Comment by slashdotdash** [comment]
@astery It's not ideal that the user provided keys are strings, whereas the system defined keys are atoms. As the metadata keys could be dynamic (whatever the user chooses) we cannot convert them to atoms, due to Erlang limits (255 chars, 1,048,576 total atoms allowed). 

One benefit of having the distinction is that it's not possible for a user to *overwrite* one of the system key/values pairs (e.g. `event_id`, `created_at`).

**Comment by astery** [comment]
Agree, we really should not enforce conversion to atoms from strings that
can be dynamic.

Slightly related another issue: if I pass an atom I expect an atom in
result.

It not bother me a lot, just thinking about ideal

**Comment by astery** [comment]
I've accidentally post at draft, I just going to post a PR that describes behavior for metadata and events serialization, just wait for it.

**Comment by slashdotdash** [comment]
Fixed by #108.
